SpotifyArtistsViewer/SARequestManager.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  SARequestManager.swift
+//  SpotifyArtistsViewer
+//
+//  Created by Kevin Zeckser on 6/6/16.
+//  Copyright © 2016 Kevin Zeckser. All rights reserved.
+//
+import Foundation
+
+enum SAResponse {
+    case Success(artists: [SAArtist])
+    case Failure(error: ErrorType)
+}
+
+
+final class SARequestManager {
+    
+    // MARK: - Properties
+    
+    static let sharedInstance = SARequestManager()
+    private let session: NSURLSession!
+    // This prevents others from using the default '()' initializer for this class.
+    private init() {
+        let sessionConfig: NSURLSessionConfiguration = NSURLSessionConfiguration.defaultSessionConfiguration()
+        session = NSURLSession(configuration: sessionConfig)
+    }
+    
+    // MARK: - NSURLRequest & JSON Parse
+    
+    func getArtistsWithQuery(query: String, completion: SAResponse -> Void) {
+        let path = "https://api.spotify.com/v1/search?q=\(query)&type=track,artist&market=US"
+        guard let url = NSURL(string: path) else { return }
+        
+        let task = session.dataTaskWithURL(url) { (data: NSData?, response: NSURLResponse?, error: NSError?) in
+            var result: SAResponse
+            
+            // We have to handle a few cases here:
+            
+            if let error = error {
+                // 1. Something went wrong with the request. We pass along the error.
+                result = SAResponse.Failure(error: error)
+            } else if let data = data {
+                
+                do {
+                    let jsonResult = try NSJSONSerialization.JSONObjectWithData(data, options: NSJSONReadingOptions.MutableContainers) as? NSDictionary
+                    var returnedArtists = [SAArtist]()
+                    
+                    if let artistResponse = jsonResult?["artists"], let artists = artistResponse["items"] as? NSArray {
+                        for entry in artists {
+                            let artistEntry = entry as? NSDictionary
+                            guard let name = artistEntry?["name"] as? String else { continue }
```

Comment:
Typically the semantic meaning for a guard statement is "ensure this state is true before we carry out some logic".

In this case, I think it would be more intuitive if there was a guard simply to ensure that an artist entry existed, and then have a traditional if statement adding any fields that are available in the json. I think the only mission-critical thing for adding to the returnedArtists array is whether or not there is an artistEntry.

Also, are you sure this is the logic you want here? If there's an artist entry but they have no images, they should be skipped?